ambry-store/src/test/java/com.github.ambry.store/BlobStoreCompactorTest.java

```diff
@@ -0,0 +1,1760 @@
+      String segmentToDeleteFrom = logSegments.get(selectedIdx);
+      MockId idToDelete = state.getIdToDeleteFromLogSegment(state.log.getSegment(segmentToDeleteFrom));
+      if (idToDelete == null) {
+        logSegments.remove(selectedIdx);
+      } else {
+        state.addDeleteEntry(idToDelete);
+        validDataSize = getValidDataSize(logSegmentsToReduceFrom, state.time.milliseconds());
+      }
+    }
+  }
+
+  /**
+   * Compacts the {@code segmentsUnderCompaction} and verifies sanity of store and data. Also verifies that no change
+   * occurred if no change was expected and vice versa. Ensures that reloading the log does not have effects on the
+   * integrity of the store and data.
+   * @param segmentsUnderCompaction the names of the log segments under compaction.
+   * @param deleteReferenceTimeMs the reference time in ms to use to decide whether deletes are valid.
+   * @param changeExpected {@code true} if compaction will cause a change in size of data. {@code false} otherwise.
+   * @throws Exception
+   */
+  private void compactAndVerify(List<String> segmentsUnderCompaction, long deleteReferenceTimeMs,
+      boolean changeExpected) throws Exception {
+    long logSegmentSizeSumBeforeCompaction = getSumOfLogSegmentEndOffsets();
+    long logSegmentCountBeforeCompaction = state.index.getLogSegmentCount();
+    long indexSegmentCountBeforeCompaction = state.index.getIndexSegments().size();
+
+    CompactionDetails details = new CompactionDetails(deleteReferenceTimeMs, segmentsUnderCompaction);
+    long expectedValidDataSize = getValidDataSize(segmentsUnderCompaction, deleteReferenceTimeMs);
+    List<String> unaffectedSegments = getUnaffectedSegments(segmentsUnderCompaction);
+    List<LogEntry> validLogEntriesInOrder = getValidLogEntriesInOrder(segmentsUnderCompaction, deleteReferenceTimeMs);
+    Set<MockId> idsInCompactedLogSegments = getIdsInSegments(segmentsUnderCompaction);
+
+    compactor = getCompactor(state.log, DISK_IO_SCHEDULER);
+    compactor.initialize(state.index);
+
+    try {
+      compactor.compact(details);
+    } finally {
+      compactor.close(0);
+    }
+
+    assertFalse("No compaction should be in progress", CompactionLog.isCompactionInProgress(tempDirStr, STORE_ID));
+    long logSegmentSizeAfterCompaction = getSumOfLogSegmentEndOffsets();
+    long logSegmentCountAfterCompaction = state.index.getLogSegmentCount();
+    long indexSegmentCountAfterCompaction = state.index.getIndexSegments().size();
+    verifyCompaction(segmentsUnderCompaction, unaffectedSegments, expectedValidDataSize, validLogEntriesInOrder,
+        idsInCompactedLogSegments, deleteReferenceTimeMs);
+
+    state.reloadLog(true);
+    verifyCompaction(segmentsUnderCompaction, unaffectedSegments, expectedValidDataSize, validLogEntriesInOrder,
+        idsInCompactedLogSegments, deleteReferenceTimeMs);
+
+    assertEquals("Sum of log segment capacities changed after reload", logSegmentSizeAfterCompaction,
```

Comment:
Unable to find the duplicate check. Could you paste the code blocks?

This one is checking that log segment capacities (and their sum) does not change because of reload after compaction. `checkVitals()` is checking that the size changed (or did not change) b/w the original size and the size after compaction.